elastic-agent-shipper.yml

```diff
@@ -7,4 +7,19 @@ tls: false
 #log level
 logging.level: debug
 logging.selectors: ["*"]
-logging.to_stderr: true
\ No newline at end of file
+logging.to_stderr: true
+
+queue:
+  test: #There is no actual "test" queue type, remove this later.
+    events: 512
+
+metrics:
```

Comment:
You may want to have this match the agent monitoring configuration, then it could just pass it through:

```yaml
agent.monitoring:
  # enabled turns on monitoring of running processes
  enabled: false
  # enables log monitoring
  logs: false
  # enables metrics monitoring
  metrics: false
  # exposes agent metrics using http, by default sockets and named pipes are used
  http:
      # enables http endpoint
      enabled: false
      # The HTTP endpoint will bind to this hostname, IP address, unix socket or named pipe.
      # When using IP addresses, it is recommended to only use localhost.
      host: localhost
      # Port on which the HTTP endpoint will bind. Default is 0 meaning feature is disabled.
      port: 6791
```

https://www.elastic.co/guide/en/fleet/current/elastic-agent-monitoring-configuration.html

Reply:
If we want to use configuration options that are not already available in the agent configuration, then we need to consider how they get passed down to us. Ideally the shipper is an implementation detail that users do not need to explicitly configure. 

It does probably make sense to have some additional settings we can control the default values of via the shipper binary's agent specification file or something though. Those should either extend the agent monitoring configuration in a way that requires no translation, or just be a separate configuration block.

Reply:
including the `agent.` prefix?

Reply:
This is fine, exactly how this should work will be naturally discovered when we integrate with the agent properly.